src/daemon/daemon.cpp

```diff
@@ -3034,10 +3095,23 @@ mp::Daemon::async_wait_for_ready_all(grpc::ServerReaderWriterInterface<Reply, Re
 
     if (server && std::is_same<Reply, StartReply>::value)
     {
+        bool write_reply{false};
```

Comment:
I kind see why you added the `write_reply` boolean variable. However, I would suggest just writing empty message into the `server` since the emptyness is handled automatically. That results in cleaner code

```
            Reply reply;
            if (config->update_prompt->is_time_to_show())
            {
                config->update_prompt->populate(reply.mutable_update_info());
            }

            if (warnings.size() > 0)
            {
                reply.set_log_line(fmt::to_string(warnings));
            }
            server->Write(reply);
```